cli/config.go

```diff
@@ -577,9 +577,51 @@ func loadConfiguration(configPath string, ignoreLogging bool) (resolvedConfigPat
 		config.ProxyConfig = vc.ProxyConfig{}
 	}
 
+	if err := checkHypervisorConfig(config.HypervisorConfig); err != nil {
+		return "", config, err
+	}
+
 	return resolved, config, nil
 }
 
+// checkHypervisorConfig performs basic "sanity checks" on the hypervisor
+// config.
+func checkHypervisorConfig(config vc.HypervisorConfig) error {
```

Comment:
While initramfs needs to be read all into the guest memory, the disk image does not need to be. So the sanity check can only apply to the `initrd` case.

And it can only be a best effort check, because we do not know how much memory the uncompressed initramfs and guest kernel will actually use.

Reply:
Right. I can change the PR to only check the image, but I wonder if we want to just comment the initrd behaviour and still perform the check. Wouldn't it still be pretty unusual / crazy to configure the VM to have less memory than the size of the initrd (particularly given initrd's are tiny? :)

Reply:
I'm fine with either way you choose :)